### PR TITLE
Feat/group location search

### PR DESCRIPTION
**Pull request by danpatrickoneill**
# Description

Added members results to location-based searches; some changes were overwritten by merge with master but were re-checked afterwards and work as intended.

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

**Comment by danpatrickoneill** [comment]
[User story: Search for groups by location](https://trello.com/c/Wb91ekel/33-user-story-search-for-groups-by-location)

File: controllers/group.js [hunk 1]
```diff
@@ -50,7 +50,20 @@ router.route("/search").post(async (req, res) => {
...
 
-      const groupByFilter = await Groups.search(req.body);
+      const groups = await Groups.search(req.body);
+      const group_id = groups.map(group => group.id);
+
+      const members = await GroupsUsers.find({ group_id });
+      const groupByFilter = groups.map(group => {
+        return {
+          ...group,
+          members: members.filter(member => member.group_id === group.id)
```
Comment (DaftBeowulf) [inline controllers/group.js]: neatly done with the filtering here

Comment (AngXuDev) [reply controllers/group.js]: Agreed - where did you get this beautiful code Dan

File: models/groups_users.js [hunk 2]
```diff
@@ -1,113 +1,113 @@
...
+  update,
+  remove
 };
 
 function add(group_user) {
-	return (
-		db("groups_users")
-			// user_id, user_type, group_id required
-			.insert(group_user, ["*"])
-			.then(g_u => find({ "g_u.id": g_u[0].id }).first())
```
Comment (DaftBeowulf) [inline models/groups_users.js]: It actually needs to be this filter key in quotes for the aliased db reference to recognize the tablename -- same down below for the `update` function's nested .find call

**Pull request merged**